### ISSUE TITLE
Wrong column name assumption on findBy

### ISSUE THREAD

**Issue by alberto-pelosi** [micronaut-projects/micronaut-data#353]
Version: Micronaut Data 1.0.0.M5 (Just updated from 1.0.0.M3)

Given an entity like: 

```
@Entity
@Table(name = "app_user")
@Data
@ValidUser(message = "Mail already exists")
public class ApplicationUser {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    @NotNull
    private String name;

    @NotEmpty(message = "{email.notempty}")
    @ValidEmail(message = "{email.invalidformat}")
    private String email;

    @JsonIgnore
    private String password;

    @NotNull
    private UserRole role;

    private Boolean enabled = true;

    @Relation(value = Relation.Kind.MANY_TO_ONE)
    @MappedProperty("company_id")
    private Company company;


}
```
and a repository like:

```
@JdbcRepository(dialect = Dialect.POSTGRES)
public abstract class ApplicationUserRepository implements CrudRepository<ApplicationUser, Long> {



   public Boolean checkUser(String email, String password) {
        return userRepository.findAllByEmailLike(email).stream().findFirst()
                .filter(u -> bce.matches(password, u.getPassword()))
                .map(u -> Boolean.TRUE)
                .orElse(Boolean.FALSE);
    }
}
```
I obtain the error:

Unexpected error occurred: Error reading object for name [_company_id] from result set: The column name _company_id was not found in this ResultSet.
io.micronaut.data.exceptions.DataAccessException: Error reading object for name [_company_id] from result set: The column name _company_id was not found in this ResultSet.

On db column name is company_id (without heading "_")

**Comment by graemerocher** [comment]
Please try the recently released 1.0.0.RC1 version

**Comment by alberto-pelosi** [comment]
1.0.0.RC1 fixes the problem.
